src/hip/hip_build_utils.cpp

```diff
@@ -102,8 +102,8 @@ static boost::filesystem::path HipBuildImpl(boost::optional<TmpDir>& tmp_dir,
                                             std::string src,
                                             std::string params,
                                             const TargetProperties& target,
-                                            const bool testing_mode,
-                                            const bool sources_already_reside_on_filesystem)
+                                            bool testing_mode,
```

Comment:
The fix by removing of `const` looks suspicious. Could you please share tidy error with me?

Reply:
See [here](http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/igemm-upstream2/28/pipeline).

> [2021-03-02T00:29:04.070Z] /var/jenkins/workspace/MLLibs_MIOpen_igemm-upstream2/src/include/miopen/hip_build_utils.hpp:44:34: error: parameter 'sources_already_reside_on_filesystem' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions 

Indeed, from a caller (header) perspective, this const is not doing anything because the it is not possible for callee to change the value of the copy-by-value boolean. I think it ask me to get rid of the const from header only, but I removed it from implementation as well to avoid confusion.

Reply:
> Hmm. I highly recommend resolving ROCmSoftwarePlatform/llvm-project-mlir#61 (review) and updating this PR.

Inlined reply here:

Sure, I'll wait our discussion to resolve there, and bump the commit id in this PR to make sure we are on the same page.

Reply:
> I think it ask me to get rid of the const from header only, but I removed it from implementation as well to avoid confusion.

Please remove it in the header only, keep implementation unchanged.

Reply:
>> error: parameter 'sources_already_reside_on_filesystem' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions

Idiotic warning.